### ISSUE TITLE
Report '\08' and '\09' in no-octal-escape

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12080]
**Tell us about your environment**

* **ESLint Version:** 6.1.0
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2015,
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/*eslint no-octal-escape: "error"*/

"\08"
"\09"

```

```bash
eslint index.js
```

**What did you expect to happen?**

2 errors

**What actually happened? Please include the actual, raw output from ESLint.**

No errors

**Are you willing to submit a pull request to fix this bug?**

Yes, I'll be glad to do it. (#12079 should be merged first as this would collide)

Engines/parsers are already reporting these as octal escapes in strict mode:

```js
"use strict";
"\08"
```

* Chrome: `Octal escape sequences are not allowed in strict mode.`
* Firefox: `"0"-prefixed octal literals and octal escape sequences are deprecated; for octal literals use the "0o" prefix instead`
* Espree: `Octal literal in strict mode`

Per the spec. `08` is not a valid [escape sequence](https://www.ecma-international.org/ecma-262/10.0/index.html#prod-EscapeSequence) but it seems that it can be interpreted like `0` as a [LegacyOctalEscapeSequence](https://www.ecma-international.org/ecma-262/10.0/index.html#prod-annexB-LegacyOctalEscapeSequence) followed by a character `8`.

> LegacyOctalEscapeSequence::
> 
>   OctalDigit[lookahead ∉ OctalDigit]

So I guess to be technically correct the rule should report `\0`?

**Comment by mdjermanovic** [comment]
Reopening because it's probably a false negative.

**Comment by kaicataldo** [comment]
Marking this as accepted since it's a bug report with three 👍s

**Comment by mdjermanovic** [comment]
Looks like no one is working on this, so I'll take it.
